### DISCUSSION TITLE
efac:Nationality

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#442] [Q&A] [answered]
Hi Lab ! 
about #440 
in sdk 1.6 efac:Nationality  it appears only one, but it gives an xsd error , because , I guess , BT-706-UBO has got "type" : "code" while NationalityIDType has got this definition : 
 <xsd:complexType name="NationalityIDType">
      <xsd:simpleContent>
         <xsd:restriction base="udt:IdentifierType"/>
      </xsd:simpleContent>
   </xsd:complexType>
  
here the xml  : 
<?xml version="1.0" encoding="UTF-8"?>
<ContractAwardNotice xmlns="urn:oasis:names:specification:ubl:schema:xsd:ContractAwardNotice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:efac="http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1" xmlns:efbc="http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1" xmlns:efext="http://data.europa.eu/p27/eforms-ubl-extensions/1" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2">
	<ext:UBLExtensions>
		<ext:UBLExtension>
			<ext:ExtensionContent>
				<efext:EformsExtension>
					<efac:NoticeResult>
						<efac:LotResult>
							<cbc:ID schemeName="result">RES-0001</cbc:ID>
							<cbc:TenderResultCode listName="winner-selection-status">selec-w</cbc:TenderResultCode>
							<efac:LotTender>
								<cbc:ID>TEN-0001</cbc:ID>
							</efac:LotTender>
							<efac:SettledContract>
								<cbc:ID>CON-0001</cbc:ID>
							</efac:SettledContract>
							<efac:TenderLot>
								<cbc:ID>LOT-0001</cbc:ID>
							</efac:TenderLot>
						</efac:LotResult>
						<efac:LotTender>
							<cbc:ID schemeName="tender">TEN-0001</cbc:ID>
							<efac:TenderingParty>
								<cbc:ID>TPA-0001</cbc:ID>
							</efac:TenderingParty>
							<efac:TenderLot>
								<cbc:ID>LOT-0001</cbc:ID>
							</efac:TenderLot>
						</efac:LotTender>
						<efac:SettledContract>
							<cbc:ID schemeName="contract">CON-0001</cbc:ID>
						</efac:SettledContract>
						<efac:TenderingParty>
							<cbc:ID schemeName="tendering-party">TPA-0001</cbc:ID>
							<efac:Tenderer>
								<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
							</efac:Tenderer>
						</efac:TenderingParty>
					</efac:NoticeResult>
					<efac:NoticeSubType>
						<cbc:SubTypeCode listName="notice-subtype">29</cbc:SubTypeCode>
					</efac:NoticeSubType>
					<efac:Organizations>
						<efac:Organization>
							<efac:UltimateBeneficialOwner>
								<cbc:ID>UBO-0001</cbc:ID>
							</efac:UltimateBeneficialOwner>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Organization del Tendere</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>ROMA</cbc:CityName>
									<cbc:CountrySubentityCode listName="nuts">AT122</cbc:CountrySubentityCode>
									<cac:Country>
										<cbc:IdentificationCode listName="country">BEL</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:PartyLegalEntity>
									<cbc:CompanyID>11111111111</cbc:CompanyID>
								</cac:PartyLegalEntity>
								<cac:Contact>
									<cbc:Telephone>05343434</cbc:Telephone>
									<cbc:ElectronicMail>sdsd@sdds.sdsd</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
						</efac:Organization>
						<efac:Organization>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0002</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Organization del Buyer</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>Vertemate con Minoprio</cbc:CityName>
									<cbc:CountrySubentityCode listName="nuts">AT125</cbc:CountrySubentityCode>
									<cac:Country>
										<cbc:IdentificationCode listName="country">ITA</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:PartyLegalEntity>
									<cbc:CompanyID>222222222222222222</cbc:CompanyID>
								</cac:PartyLegalEntity>
								<cac:Contact>
									<cbc:Telephone>061112111</cbc:Telephone>
									<cbc:ElectronicMail>buyere@hmails.ce</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
							<efac:TouchPoint>
								<cac:PartyIdentification>
									<cbc:ID schemeName="touchpoint">TPO-0002</cbc:ID>
								</cac:PartyIdentification>
							</efac:TouchPoint>
						</efac:Organization>
						<efac:UltimateBeneficialOwner>
							<cbc:ID schemeName="ubo">UBO-0001</cbc:ID>
							<cbc:FamilyName>Ferdinando</cbc:FamilyName>
							<cac:ResidenceAddress>
								<cbc:AdditionalStreetName>40141</cbc:AdditionalStreetName>
								<cbc:CityName>PISA</cbc:CityName>
								<cbc:CountrySubentityCode listName="nuts">AL033</cbc:CountrySubentityCode>
								<cac:Country>
									<cbc:IdentificationCode listName="country">ITA</cbc:IdentificationCode>
								</cac:Country>
							</cac:ResidenceAddress>
							<efac:Nationality>
								<cbc:NationalityID listName="country">ITA</cbc:NationalityID>
							</efac:Nationality>
						</efac:UltimateBeneficialOwner>
					</efac:Organizations>
				</efext:EformsExtension>
			</ext:ExtensionContent>
		</ext:UBLExtension>
	</ext:UBLExtensions>
	<cbc:UBLVersionID>2.3</cbc:UBLVersionID>
	<cbc:CustomizationID>eforms-sdk-1.6</cbc:CustomizationID>
	<cbc:ID schemeName="notice-id">955bf7d9-843a-4194-b7e8-4835e2161385</cbc:ID>
	<cbc:ContractFolderID>c2a0dcd1-4d17-4072-9252-38de16ead921</cbc:ContractFolderID>
	<cbc:IssueDate>2023-03-03+01:00</cbc:IssueDate>
	<cbc:IssueTime>01:06:21+02:00</cbc:IssueTime>
	<cbc:VersionID>01</cbc:VersionID>
	<cbc:RegulatoryDomain>31985R2137</cbc:RegulatoryDomain>
	<cbc:NoticeTypeCode listName="result">can-standard</cbc:NoticeTypeCode>
	<cbc:NoticeLanguageCode listName="language">ENG</cbc:NoticeLanguageCode>
	<cac:ContractingParty>
		<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-legal-type">body-pl-ra</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
		<cac:ContractingActivity>
			<cbc:ActivityTypeCode listName="authority-activity">hc-am</cbc:ActivityTypeCode>
		</cac:ContractingActivity>
		<cac:Party>
			<cac:PartyIdentification>
				<cbc:ID schemeName="organization">ORG-0002</cbc:ID>
			</cac:PartyIdentification>
		</cac:Party>
	</cac:ContractingParty>
	<cac:TenderingProcess>
		<cbc:ProcedureCode listName="procurement-procedure-type">comp-tend</cbc:ProcedureCode>
	</cac:TenderingProcess>
	<cac:ProcurementProject>
		<cbc:Name languageID="ENG">PROC TITLE</cbc:Name>
		<cbc:Description languageID="ENG">PROC_DESC</cbc:Description>
		<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
		<cac:RequestedTenderTotal>
			<cbc:EstimatedOverallContractAmount currencyID="EUR">100</cbc:EstimatedOverallContractAmount>
		</cac:RequestedTenderTotal>
		<cac:MainCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode>
		</cac:MainCommodityClassification>
		<cac:AdditionalCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode>
		</cac:AdditionalCommodityClassification>
	</cac:ProcurementProject>
	<cac:ProcurementProjectLot>
		<cbc:ID schemeName="Lot">LOT-0001</cbc:ID>
		<cac:TenderingTerms>
			<cbc:FundingProgramCode listName="eu-funded">eu-funds</cbc:FundingProgramCode>
			<cac:AwardingTerms>
				<cac:AwardingCriterion>
					<cac:SubordinateAwardingCriterion>
						<cbc:AwardingCriterionTypeCode listName="award-criterion-type">cost</cbc:AwardingCriterionTypeCode>
					</cac:SubordinateAwardingCriterion>
				</cac:AwardingCriterion>
			</cac:AwardingTerms>
			<cac:AppealTerms>
				<cac:AppealReceiverParty>
					<cac:PartyIdentification>
						<cbc:ID schemeName="touchpoint">TPO-0002</cbc:ID>
					</cac:PartyIdentification>
				</cac:AppealReceiverParty>
			</cac:AppealTerms>
		</cac:TenderingTerms>
		<cac:TenderingProcess>
			<cbc:GovernmentAgreementConstraintIndicator>true</cbc:GovernmentAgreementConstraintIndicator>
			<cac:AuctionTerms>
				<cbc:AuctionConstraintIndicator>true</cbc:AuctionConstraintIndicator>
			</cac:AuctionTerms>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="framework-agreement">fa-mix</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="dps-usage">none</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
		</cac:TenderingProcess>
		<cac:ProcurementProject>
			<cbc:Name languageID="ENG">LOT TITLE</cbc:Name>
			<cbc:Description languageID="ENG">LOT DESCR</cbc:Description>
			<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
			<cac:RequestedTenderTotal>
				<cbc:EstimatedOverallContractAmount currencyID="EUR">250</cbc:EstimatedOverallContractAmount>
			</cac:RequestedTenderTotal>
			<cac:MainCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode>
			</cac:MainCommodityClassification>
			<cac:AdditionalCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode>
			</cac:AdditionalCommodityClassification>
		</cac:ProcurementProject>
	</cac:ProcurementProjectLot>
	<cac:TenderResult>
		<cbc:AwardDate>2023-05-05+02:00</cbc:AwardDate>
	</cac:TenderResult>
</ContractAwardNotice>

Thanks and regards
Renato

**Comment by rouschr** [comment]
Can you please repost the XML and put it inside of the rich text <> (code)

**Reply by rengit62** [reply]
```
<?xml version="1.0" encoding="UTF-8"?>
<ContractAwardNotice xmlns="urn:oasis:names:specification:ubl:schema:xsd:ContractAwardNotice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:efac="http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1" xmlns:efbc="http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1" xmlns:efext="http://data.europa.eu/p27/eforms-ubl-extensions/1" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2">
	<ext:UBLExtensions>
		<ext:UBLExtension>
			<ext:ExtensionContent>
				<efext:EformsExtension>
					<efac:NoticeResult>
						<efac:LotResult>
							<cbc:ID schemeName="result">RES-0001</cbc:ID>
							<cbc:TenderResultCode listName="winner-selection-status">selec-w</cbc:TenderResultCode>
							<efac:LotTender>
								<cbc:ID>TEN-0001</cbc:ID>
							</efac:LotTender>
							<efac:SettledContract>
								<cbc:ID>CON-0001</cbc:ID>
							</efac:SettledContract>
							<efac:TenderLot>
								<cbc:ID>LOT-0001</cbc:ID>
							</efac:TenderLot>
						</efac:LotResult>
						<efac:LotTender>
							<cbc:ID schemeName="tender">TEN-0001</cbc:ID>
							<efac:TenderingParty>
								<cbc:ID>TPA-0001</cbc:ID>
							</efac:TenderingParty>
							<efac:TenderLot>
								<cbc:ID>LOT-0001</cbc:ID>
							</efac:TenderLot>
						</efac:LotTender>
						<efac:SettledContract>
							<cbc:ID schemeName="contract">CON-0001</cbc:ID>
						</efac:SettledContract>
						<efac:TenderingParty>
							<cbc:ID schemeName="tendering-party">TPA-0001</cbc:ID>
							<efac:Tenderer>
								<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
							</efac:Tenderer>
						</efac:TenderingParty>
					</efac:NoticeResult>
					<efac:NoticeSubType>
						<cbc:SubTypeCode listName="notice-subtype">29</cbc:SubTypeCode>
					</efac:NoticeSubType>
					<efac:Organizations>
						<efac:Organization>
							<efac:UltimateBeneficialOwner>
								<cbc:ID>UBO-0001</cbc:ID>
							</efac:UltimateBeneficialOwner>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Organization del Tendere</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>ROMA</cbc:CityName>
									<cbc:CountrySubentityCode listName="nuts">AT122</cbc:CountrySubentityCode>
									<cac:Country>
										<cbc:IdentificationCode listName="country">BEL</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:PartyLegalEntity>
									<cbc:CompanyID>11111111111</cbc:CompanyID>
								</cac:PartyLegalEntity>
								<cac:Contact>
									<cbc:Telephone>05343434</cbc:Telephone>
									<cbc:ElectronicMail>sdsd@sdds.sdsd</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
						</efac:Organization>
						<efac:Organization>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0002</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Organization del Buyer</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>Vertemate con Minoprio</cbc:CityName>
									<cbc:CountrySubentityCode listName="nuts">AT125</cbc:CountrySubentityCode>
									<cac:Country>
										<cbc:IdentificationCode listName="country">ITA</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:PartyLegalEntity>
									<cbc:CompanyID>222222222222222222</cbc:CompanyID>
								</cac:PartyLegalEntity>
								<cac:Contact>
									<cbc:Telephone>061112111</cbc:Telephone>
									<cbc:ElectronicMail>buyere@hmails.ce</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
							<efac:TouchPoint>
								<cac:PartyIdentification>
									<cbc:ID schemeName="touchpoint">TPO-0002</cbc:ID>
								</cac:PartyIdentification>
							</efac:TouchPoint>
						</efac:Organization>
						<efac:UltimateBeneficialOwner>
							<cbc:ID schemeName="ubo">UBO-0001</cbc:ID>
							<cbc:FamilyName>Ferdinando</cbc:FamilyName>
							<cac:ResidenceAddress>
								<cbc:AdditionalStreetName>40141</cbc:AdditionalStreetName>
								<cbc:CityName>PISA</cbc:CityName>
								<cbc:CountrySubentityCode listName="nuts">AL033</cbc:CountrySubentityCode>
								<cac:Country>
									<cbc:IdentificationCode listName="country">ITA</cbc:IdentificationCode>
								</cac:Country>
							</cac:ResidenceAddress>
							<efac:Nationality>
								<cbc:NationalityID listName="country">ITA</cbc:NationalityID>
							</efac:Nationality>
						</efac:UltimateBeneficialOwner>
					</efac:Organizations>
				</efext:EformsExtension>
			</ext:ExtensionContent>
		</ext:UBLExtension>
	</ext:UBLExtensions>
	<cbc:UBLVersionID>2.3</cbc:UBLVersionID>
	<cbc:CustomizationID>eforms-sdk-1.6</cbc:CustomizationID>
	<cbc:ID schemeName="notice-id">955bf7d9-843a-4194-b7e8-4835e2161385</cbc:ID>
	<cbc:ContractFolderID>c2a0dcd1-4d17-4072-9252-38de16ead921</cbc:ContractFolderID>
	<cbc:IssueDate>2023-03-03+01:00</cbc:IssueDate>
	<cbc:IssueTime>01:06:21+02:00</cbc:IssueTime>
	<cbc:VersionID>01</cbc:VersionID>
	<cbc:RegulatoryDomain>31985R2137</cbc:RegulatoryDomain>
	<cbc:NoticeTypeCode listName="result">can-standard</cbc:NoticeTypeCode>
	<cbc:NoticeLanguageCode listName="language">ENG</cbc:NoticeLanguageCode>
	<cac:ContractingParty>
		<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-legal-type">body-pl-ra</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
		<cac:ContractingActivity>
			<cbc:ActivityTypeCode listName="authority-activity">hc-am</cbc:ActivityTypeCode>
		</cac:ContractingActivity>
		<cac:Party>
			<cac:PartyIdentification>
				<cbc:ID schemeName="organization">ORG-0002</cbc:ID>
			</cac:PartyIdentification>
		</cac:Party>
	</cac:ContractingParty>
	<cac:TenderingProcess>
		<cbc:ProcedureCode listName="procurement-procedure-type">comp-tend</cbc:ProcedureCode>
	</cac:TenderingProcess>
	<cac:ProcurementProject>
		<cbc:Name languageID="ENG">PROC TITLE</cbc:Name>
		<cbc:Description languageID="ENG">PROC_DESC</cbc:Description>
		<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
		<cac:RequestedTenderTotal>
			<cbc:EstimatedOverallContractAmount currencyID="EUR">100</cbc:EstimatedOverallContractAmount>
		</cac:RequestedTenderTotal>
		<cac:MainCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode>
		</cac:MainCommodityClassification>
		<cac:AdditionalCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode>
		</cac:AdditionalCommodityClassification>
	</cac:ProcurementProject>
	<cac:ProcurementProjectLot>
		<cbc:ID schemeName="Lot">LOT-0001</cbc:ID>
		<cac:TenderingTerms>
			<cbc:FundingProgramCode listName="eu-funded">eu-funds</cbc:FundingProgramCode>
			<cac:AwardingTerms>
				<cac:AwardingCriterion>
					<cac:SubordinateAwardingCriterion>
						<cbc:AwardingCriterionTypeCode listName="award-criterion-type">cost</cbc:AwardingCriterionTypeCode>
					</cac:SubordinateAwardingCriterion>
				</cac:AwardingCriterion>
			</cac:AwardingTerms>
			<cac:AppealTerms>
				<cac:AppealReceiverParty>
					<cac:PartyIdentification>
						<cbc:ID schemeName="touchpoint">TPO-0002</cbc:ID>
					</cac:PartyIdentification>
				</cac:AppealReceiverParty>
			</cac:AppealTerms>
		</cac:TenderingTerms>
		<cac:TenderingProcess>
			<cbc:GovernmentAgreementConstraintIndicator>true</cbc:GovernmentAgreementConstraintIndicator>
			<cac:AuctionTerms>
				<cbc:AuctionConstraintIndicator>true</cbc:AuctionConstraintIndicator>
			</cac:AuctionTerms>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="framework-agreement">fa-mix</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="dps-usage">none</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
		</cac:TenderingProcess>
		<cac:ProcurementProject>
			<cbc:Name languageID="ENG">LOT TITLE</cbc:Name>
			<cbc:Description languageID="ENG">LOT DESCR</cbc:Description>
			<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
			<cac:RequestedTenderTotal>
				<cbc:EstimatedOverallContractAmount currencyID="EUR">250</cbc:EstimatedOverallContractAmount>
			</cac:RequestedTenderTotal>
			<cac:MainCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode>
			</cac:MainCommodityClassification>
			<cac:AdditionalCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode>
			</cac:AdditionalCommodityClassification>
		</cac:ProcurementProject>
	</cac:ProcurementProjectLot>
	<cac:TenderResult>
		<cbc:AwardDate>2023-05-05+02:00</cbc:AwardDate>
	</cac:TenderResult>
</ContractAwardNotice>
```

**Comment by rouschr** [comment, marked as answer]
You can see the details here for SDK 1.6:
https://github.com/OP-TED/eForms-SDK/blob/1.6.0/fields/fields.json
"ND-UBO" is repeatable
"BT-706-UBO" is repeatable (efac:Nationality/cbc:NationalityID)
The problem here is that efac:Nationality was made part of the field relative xpath.
In terms of rules on our side we meant that it can exist more than once in the XML.
But the field as an XML element is not really repeating inside of the parent element.

To solve this in 1.8 we will have a new node:
"ND-UBONationality" repeatable (efac:Nationality)
"BT-706-UBO" not repeatable (cbc:NationalityID)
This should solve XSD validation issues around field repeatability.
